### PR TITLE
Initial Skeleton

### PR DESCRIPTION
**Pull request by Dermah**
Sets an outline of how the script will work with some unimplemented tests that describe important functionality. 

Things that could be bettered:

- [x] Setup guide
- [x] Run guide
- [x] API Documentation
- [x] High level description

### CODE REVIEW

**Comment by mrjackdavis** [comment]
@Dermah some newbie friendly setup & run documentation would go a fair way

**Comment by mrjackdavis** [comment]
Actually, and some rough documentation of the API. So that people new to the repo can understand what it's doing at a high level

**Comment by Dermah** [comment]
Good idea, I'll make a task list here or somewhere to track this

- [ ] Setup guide
- [ ] Run guide
- [ ] API Documentation
- [ ] High level description

File: test/_TestEC2Store.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			mockEC2.describeSnapshots = sinon.stub().yields(null, ec2Responses.snapshots1);
+
+			return ec2Store.listSnapshots()
+				.then(snapList => {
+					expect(mockEC2.describeSnapshots.called).to.be.ok();
+					return;
+				});
+		});
+		it('should return a Promise that resolves to an array of EC2 snapshots');
+		it('should filter snapshots so that only snapshots owned by us ... something')
```
Comment (mrjackdavis) [inline test/_TestEC2Store.js]: ?

File: test/_TestEC2Store.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		it('should ask AWS EC2 for a list of all snapshots', () => {
+			mockEC2.describeSnapshots = sinon.stub().yields(null, ec2Responses.snapshots1);
+
+			return ec2Store.listSnapshots()
+				.then(snapList => {
+					expect(mockEC2.describeSnapshots.called).to.be.ok();
+					return;
+				});
+		});
+		it('should return a Promise that resolves to an array of EC2 snapshots');
```
Comment (mrjackdavis) [inline test/_TestEC2Store.js]: I feel like this and the above test will have pretty much identical implementation code. Why not wrap them in a description and have a before function to do the work?

**Comment by mrjackdavis** [comment]
Should we add `.skip` to all unimplemented tests? It would help with visibility of completion

File: README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+```
+This runs the [`index.js`](index.js) file, which uses the built code from the previous step. In the beginning this won't do anything useful since most of the code hasn't been written yet. Running may change later once the code is closer to production stage.
+
+## Backup Runner API
+
+The function exported by [`src/index.js`](src/index.js) should run the entire backup process (described in 'What does this do?' section above) using the APIs provided by the modules in `src/`. For detail on these APIs, see the [API doc](docs/API.md)
+
+## Backup Tag API
+
+_To be finialised_
```
Comment (mrjackdavis) [inline README.md]: I feel like this is important to finalise now

File: package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "main": "index.js",
+  "directories": {
+    "test": "tests"
+  },
+  "scripts": {
+    "test": "npm run lint && node ./node_modules/mocha/bin/mocha --compilers js:babel-register",
+    "lint": "node ./node_modules/eslint/bin/eslint.js index.js src/**",
+    "build": "babel src -d build --source-maps",
+    "start": "node index.js"
+  },
```
Comment (mrjackdavis) [inline package.json]: (y)

File: test/_TestEC2Store.js (hunk 1 above)
Comment (Dermah) [inline test/_TestEC2Store.js]: Couldn't figure out how to word this one, basically if you do no filtering when asking for snapshots then you get public snapshots by other companies as well. So we need to filter the snapshots returned by ones that are owned by the current account or maybe an account id parameter or something.

Comment (mrjackdavis) [inline test/_TestEC2Store.js]: Hm, 'should exclusively return snapshots owned by current account'?

**Comment by mrjackdavis** [comment]
Following the steps in the readme, I was able to get this working on node `v5.1.0`

**Comment by mrjackdavis** [comment]
@Dermah :white_check_mark: LGTM

**Pull request merged**